src/controller/CommissioningWindowOpener.cpp

```diff
@@ -126,6 +126,10 @@ CHIP_ERROR CommissioningWindowOpener::OpenCommissioningWindow(const Commissionin
     mCommissioningWindowVerifierCallback = nullptr;
     mNodeId                              = params.GetNodeId();
     mCommissioningWindowTimeout          = params.GetTimeout();
+    if (params.HasEndpointId())
+    {
+        mTargetEndpointId = params.GetEndpointId();
```

Comment:
This looks wrong.  A CommissioningWindowOpener object can be reused, and this is leaking state from the previous params, no?

The right thing to do is that either this code resets to kRootEndpointId, or the params return that if they have nothing else specified.  The latter seems nicer to me, since the params can just always have an endpoint id, defaulting to kRootEndpointId.

Reply:
Great catch! The params now default to this: https://github.com/project-chip/connectedhomeip/pull/34425/commits/badd7f4e6f357476d446ea5d01910b4faa50c55b